test/suite/test_compact07.py

```diff
@@ -169,8 +168,15 @@ def test_compact07(self):
         self.assertGreater(success, 0)
         stat_cursor.close()
 
-        # FIXME-WT-11432: Background and foreground compaction should not run in parallel, stop the
-        # background compaction server before proceeding.
+        # Perform foreground compaction on the remaining file by setting a free_space_target value
+        # that is guaranteed to run on it. This call might return EBUSY if background compaction is
+        # inspecting it at the same time.
+        self.compactUntilSuccess(self.session, uri_small, 'free_space_target=1MB')
+
+        # Check that foreground compaction has done some work on the small table.
+        self.assertGreater(self.get_pages_rewritten(uri_small), 0)
```

Comment:
The concurrent behavior can be tricky to test and it is hard to control what the background compaction is trying to compact when we get to that line. We could have only one table to increase the chances of concurrent behaviour.
In this test, since the background compaction is configured with a threshold that is too big for that table, it will only go through the estimate phase and quit which makes the time window very limited. After WT-11343, the background compaction will look at that table even less.

In terms of better testing, we are relying on test/format to test that behaviour randomly every now and then and on WT-11346 where we will stress parallel compaction.

Reply:
It seems that one of our tests was triggering concurrent foreground compaction, I had to revert those changes and create a new PR, see #9565.